internal/ledgerutil/compare.go

```diff
@@ -428,19 +432,29 @@ func earlierSSRecord(r1 *snapshotRecord, r2 *snapshotRecord) *snapshotRecord {
 }
 
 // Creates a new SnapshotRecord
-func newSnapshotRecord(record *privacyenabledstate.SnapshotRecord) (*snapshotRecord, error) {
+func newSnapshotRecord(record *privacyenabledstate.SnapshotRecord, hashed bool) (*snapshotRecord, error) {
 	blockNum, txNum, err := heightFromBytes(record.Version)
 	if err != nil {
 		return nil, err
 	}
 
 	return &snapshotRecord{
-		Value:    string(record.Value),
+		Value:    bytesToString(record.Value, hashed),
 		BlockNum: blockNum,
 		TxNum:    txNum,
 	}, nil
 }
 
+// Converts byte slice to string, respects hashed data integrity
+// If data is hashsed, converts byte slice to hexadecimal string encoding
+func bytesToString(v []byte, h bool) string {
```

Comment:
isn't it easier to just use `string(v)` explicitly instead of calling a function and passing it false?

Reply:
ok and here is another question/proposal: 

Why not use: [utf8.Valid](https://pkg.go.dev/unicode/utf8#Valid) to determine if it's a valid UTF8 encoding and then [print as hex based on that](https://go.dev/play/p/p6Ke0RgpeTb)? Then you won't need to "drag" the hashed boolean all the way down.

```
package main

import (
	"encoding/hex"
	"fmt"
	"unicode/utf8"
)

func asReadableString(in []byte) string {
	if utf8.Valid(in) {
		return string(in)
	}
	return fmt.Sprintf("0x%s", hex.EncodeToString(in))
}

func main() {
	realString := []byte("Hello, 世界")
	someGarbage := []byte{0xff, 0xfe, 0xfd}
	someGarbageAsHex := []byte(hex.EncodeToString(someGarbage))

	fmt.Println(asReadableString(realString))       // prints Hello, 世界
	fmt.Println(asReadableString(someGarbage))      // prints 0xfffefd
	fmt.Println(asReadableString(someGarbageAsHex)) // prints fffefd
}

```

Reply:
This is also an option. I thought it'd be a little more uniform and less confusing to simply convert all hashed data to hex since converting it to UTF-8 string, even if it has a valid UTF-8 encoding, doesn't really provide any extra context to the user; it's still hashed data. If we use the convention of converting hashed data to hex and public data to regular UTF-8 string we at least give the user another way to differentiate between public and private data.